travelnet.lua

```diff
@@ -1,148 +1,32 @@
-			local meta = minetest.get_meta(pos);
-			travelnet.reset_formspec( meta );
-			meta:set_string("owner", placer:get_player_name());
-			local top_pos = {x=pos.x, y=pos.y+1, z=pos.z}
-			minetest.set_node(top_pos, {name="travelnet:hidden_top"})
-	  end,
-
-	  on_receive_fields = travelnet.on_receive_fields,
-	  on_punch = function(pos, node, puncher)
-			travelnet.update_formspec(pos, puncher:get_player_name(), nil)
-	  end,
-
-	  can_dig = function( pos, player )
-			return travelnet.can_dig( pos, player, 'travelnet box' )
-	  end,
-
-	  after_dig_node = function(pos, oldnode, oldmetadata, digger)
-			travelnet.remove_box( pos, oldnode, oldmetadata, digger )
-	  end,
-
-	  -- TNT and overenthusiastic DMs do not destroy travelnets
-	  on_blast = function(pos, intensity)
-	  end,
-
-	  -- taken from VanessaEs homedecor fridge
-	  on_place = function(itemstack, placer, pointed_thing)
-
-	     local pos = pointed_thing.above;
-	     local node = minetest.get_node({x=pos.x, y=pos.y+1, z=pos.z})
-	     local def = minetest.registered_nodes[node.name]
-	   -- leftover top nodes can be removed by placing a new travelnet underneath
-	     if (not def or not def.buildable_to) and node.name ~= "travelnet:hidden_top" then
-
-	        minetest.chat_send_player(
-						placer:get_player_name(),
-						S('Not enough vertical space to place the travelnet box!')
-					)
-	        return;
-	     end
-	     return minetest.item_place(itemstack, placer, pointed_thing);
-	  end,
-
-		on_destruct = function(pos)
-	    pos = {x=pos.x, y=pos.y+1, z=pos.z}
-	    minetest.remove_node(pos)
-	  end
-	})
-
-	if cfg.recipe then
-		-- normal recipe
-		minetest.register_craft({
-			output = cfg.nodename,
-			recipe = cfg.recipe,
-		})
-	elseif has_dye_mod and cfg.dye then
-		-- dye recipe with "default" box
-		minetest.register_craft({
-			output = cfg.nodename,
-			type = "shapeless",
-			recipe = {"group:travelnet", "dye:" .. cfg.dye},
-		})
-	end
-end
 
 -- "basic" travelnet box in yellow
-register_travelnet_box({
+travelnet.register_travelnet_box({
 	nodename = "travelnet:travelnet",
 	recipe = travelnet.travelnet_recipe,
-	color = "^[multiply:#ffff00"
+	color = "#ffff00"
 })
 
-register_travelnet_box({
+travelnet.register_travelnet_box({
 	nodename = "travelnet:travelnet_red",
-	color = "^[multiply:#ff0000",
-	dye = "red"
+	color = "#ff0000",
+	dye = "dye:red"
 })
 
-register_travelnet_box({
+travelnet.register_travelnet_box({
 	nodename = "travelnet:travelnet_blue",
-	color = "^[multiply:#0000ff",
-	dye = "blue"
+	color = "#0000ff",
+	dye = "dye:blue"
 })
 
-register_travelnet_box({
+travelnet.register_travelnet_box({
 	nodename = "travelnet:travelnet_green",
-	color = "^[multiply:#00ff00",
-	dye = "green"
+	color = "#00ff00",
+	dye = "dye:green"
 })
 
-register_travelnet_box({
+travelnet.register_travelnet_box({
 	nodename = "travelnet:travelnet_black",
-	color = "^[multiply:#000000",
-	dye = "black",
+	color = "#000000",
+	dye = "dye:black",
 	light_source = 0
 })
```

Comment:
```suggestion
})

travelnet.register_travelnet_box({
	nodename = "travelnet:travelnet_white",
	color = "#000000",
	dye = "dye:white",
	light_source = 14
})
```
I think we need white too 😉

Reply:
```lua
	color = "#000000",
```
_white_ :thinking: